dts/arm/nxp/nxp_k6x.dtsi

```diff
@@ -65,6 +65,34 @@
 
 			flash0: flash@0 {
 				reg = <0 0x100000>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boot_partition: partition@0 {
+						label = "mcuboot";
+						reg = <0x00000000 0x00010000>;
+						read-only;
+					};
+					app_partition: partition@10000 {
+						label = "application";
```

Comment:
Maybe call this application-state, to make sure people don't think it's a binary?